### PR TITLE
ddssff 8.8.1

### PR DESCRIPTION
**Pull request by phadej**
A commit on top of #72. `haskell-ci` is a new name of hvr's multi-ghc-travis "script".

Particularly, it uses `v2-build`, so it "works" with GHC-8.8

### CODE REVIEW

**Comment by phadej** [comment]
Note: 8.8.1 job fails for a reason. I tried to fix the SafeCopy module, but Derive is trickier.

I'd suggest using https://hackage.haskell.org/package/th-abstraction, it helps stay sane with TH in simple scenarious. If it's ok, I can give it a try next week (23.9 ->)

File: src/Data/SafeCopy/SafeCopy.hs [hunk 1]
```diff
@@ -548,10 +540,18 @@ validChain a_proxy =
...
+checkConsistency :: (SafeCopy a, Fail.MonadFail m) => Proxy a -> m b -> m b
 checkConsistency proxy ks
     = case consistentFromProxy proxy of
-        NotConsistent msg -> fail msg
+        NotConsistent msg -> Fail.fail msg
+        Consistent        -> ks
+
+-- | PutM doesn't have reasonable 'fail' implementation.
+-- It just throws unpure exception anyway.
+unpureCheckConsistency :: SafeCopy a => Proxy a -> b -> b
```
Comment (phadej) [inline src/Data/SafeCopy/SafeCopy.hs]: Note this schange.

**Comment by ddssff** [comment]
Can we add 8.8.1 to the allowed failures list for now?

**Comment by phadej** [comment]
@ddssff done

**Comment by ddssff** [comment]
haha I'm glad I didn't have to learn how to do that

**Comment by phadej** [comment]
@ddssff `haskell-ci --help` lists most of the options (incl `--allow-failures`).

**Pull request merged**